backend/src/test/controller/PromotionController.test.ts

```diff
@@ -182,6 +242,34 @@ describe('Unit tests for PromotionController', function () {
       });
   });
 
+  test('POST /promotions then GET /promotions/:id', async (done) => {
+    const user: User = new UserFactory().generate();
+    const expectedPromotion = new PromotionFactory().generate(
+      user,
+      new DiscountFactory().generate(DiscountType.PERCENTAGE),
+      [new ScheduleFactory().generate()]
+    );
+    expectedPromotion.lat = -34.5;
+    expectedPromotion.lon = 564.321;
+
+    await userRepository.save(user);
+    request(app)
+      .post('/promotions')
+      .send({ ...expectedPromotion, user: undefined, userId: user.id })
+      .then(async () => {
+        await promotionRepository.save(expectedPromotion);
+        request(app)
+          .get(`/promotions/${expectedPromotion.id}`)
+          .expect(200)
+          .end((err, res) => {
+            if (err) return done(err);
+            const promotion = res.body;
+            comparePromotions(promotion, expectedPromotion);
```

Comment:
Ohya and I think we should update `comparePromotions` to check that the lat/lon are being returned